lib/segment/src/index/field_index/full_text_index/inverted_index.rs

```diff
@@ -506,22 +530,30 @@ impl ImmutableInvertedIndex {
                 }
             })
     }
+
+    fn vocab_with_positngs_len_iter(&self) -> impl Iterator<Item = (&str, usize)> + '_ {
+        self.vocab.iter().filter_map(|(token, &posting_idx)| {
+            if let Some(Some(postings)) = self.postings.get(posting_idx as usize) {
+                Some((token.as_str(), postings.len()))
+            } else {
+                None
+            }
+        })
+    }
 }
 
 impl From<MutableInvertedIndex> for ImmutableInvertedIndex {
     fn from(mut index: MutableInvertedIndex) -> Self {
-        index
+        let mut postings: Vec<Option<CompressedPostingList>> = index
             .postings
-            .iter_mut()
-            .filter_map(|posting| posting.as_mut())
-            .for_each(|posting| {
-                posting.shrink_to_fit();
-            });
-        index.postings.shrink_to_fit();
+            .into_iter()
+            .map(|x| x.map(CompressedPostingList::new))
+            .collect();
+        postings.shrink_to_fit();
```

Comment:
Why is `shrink_to_fit` necessary?
I'd assume the `collect` method would size the vector appropriately.

Reply:
Yes, assuming size information of the iterator is exact it is unnecessary.